torch/csrc/jit/codegen/cuda/lower_predicate_elimination.cpp

```diff
@@ -294,6 +295,47 @@ class PredicateChcker : public IterVisitor {
       return true;
     }
 
+    // TODO: This is directed WAR on FusionPersistentNormLocalShared.
+    //  This use case along with other previous issues motivate a
+    //   joint optimization of predicate removal and buffer reuse.
+    // In this particular case:
+    //   __shared__ T0 [10], T1[10]
+    //   for i in ...
+    //      T1[i] = T0[i] + ...  // exp0
+    //      T2 = 0;              // init for exp1
+    //      if(pred)
+    //        T2 = T1 ...        // exp1
+    //  If we remove pred around expr1, as the way the pred removal
+    //    pass is set up, the init for expr will be pushed up to
+    //    initialize T1 instead.
+    //  However if we initialize T1, the code will look like:
+    //  for i in ...
+    //    T1[i] = 0;
+    //  for i in ...
+    //    if(pred)
```

Comment:
Why is `T1`  predicated now?

Reply:
Good catch. There's no change in predicate around T1 in this case. Updated comment. Thanks.